runtime/runtime.m

```diff
@@ -270,8 +270,10 @@
 	if (index == -1) {
 		p = mono_signature_get_return_type (msig);
 	} else {
-		for (int i = 0; i < index + 1; i++)
+		for (int i = 0; i < index + 1; i++) {
+			xamarin_mono_object_release (&p);
```

Comment:
I assume it's fine if the first iteration points to `NULL`, right ?

Reply:
Correct, a pointer to NULL will be ignored.